src/App.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react'
+import './App.css'
+
+import * as Tone from 'tone'
+
+function App() {
+  //create a synth and connect it to the master output (your speakers)
+  const [synthPreset, setSynthPreset] = useState({
+    on: false,
+    adsr: [0, 0, 0, 0]
+  })
+  const synth = synthPreset.on
+    ? new Tone.Synth({
+        oscillator: {
+          type: 'pwm',
+          modulationFrequency: 0.2
+        },
+        envelope: {
+          attack: synthPreset.adsr[0],
+          decay: synthPreset.adsr[1],
+          sustain: synthPreset.adsr[2],
+          release: synthPreset.adsr[3]
+        }
+      }).toMaster()
+    : new Tone.Synth().toMaster()
```

Comment:
There's "probably" an alter existing synth method that could be used here? so you'd remove the synth from the react element, and then alter it as needed. Could be cleaner if that was the case, less processing overhead, as you only regenerate the synth when the button is clicked instead of on re-render